data-prepper-plugins/kafka-connect-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafkaconnect/extension/KafkaConnectConfig.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafkaconnect.extension;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.opensearch.dataprepper.plugins.kafka.configuration.AuthConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.AwsConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.EncryptionConfig;
+import org.opensearch.dataprepper.plugins.kafka.util.KafkaClusterAuthConfig;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+
+public class KafkaConnectConfig implements KafkaClusterAuthConfig {
+    private static final long CONNECTOR_TIMEOUT_MS = 30000L; // 30 seconds
+    private static final long CONNECT_TIMEOUT_MS = 60000L; // 60 seconds
+
+    @JsonProperty("worker_properties")
+    private WorkerProperties workerProperties = new WorkerProperties();
+
+    @JsonProperty("connect_timeout_ms")
```

Comment:
Data Prepper is making duration a first-class primitive in configurations to allow for a few different ways to configure: `30s`, `30000ms`, or even a full ISO-8601 string `PT30S`.

Please rename to `connection_timeout` and use a Java 8 `Duration` type here.

See this PR for an example: #3099

Reply:
Sure, good to know

Reply:
The name should be connect_start_timeout. It means how long should we wait for Kafka Connect in running state.